packages/jupyterlab-manager/src/manager.ts

```diff
@@ -111,31 +153,112 @@ class WidgetManager extends ManagerBase<Widget> implements IDisposable {
     }
   }
 
+  _handleKernelStatusChange(args: Kernel.Status) {
+    switch (args) {
+    case 'connected':
+      // TODO: should we clear away any old widgets before restoring?
```

Comment:
Did we answer this? If so, we should change this comment to explain what we chose (and why).

Reply:
Should we also reset `this._restoredStatus` here?

Reply:
Is this the only thing left to resolve here?

Reply:
I think so. I.e. both the TODO, and reviewing if `this._restoredStatus` should ever be reset (and if so, when and where).

Reply:
I went ahead and reset the restored status in disconnect. I think it makes most sense to reset it there, since a reconnect will then restore widgets again.

Reply:
Was this actually done? Since you force pushed, the commit log doesn't show any partial diffs for this, so its super hard to track what was changed.

Reply:
Sorry, and in general this PR was already hard to track being so long and involved. Are you asking if the TODO was fixed, or if the restored status was reset?